deploy/ui/form.json

```
@@ -107,26 +107,78 @@
+            }
+          ],
           "fields": [
             {
-              "label": "Username",
-              "default": "adminUser",
+              "label": "Username and password",
+              "type": "fieldGroup",
               "required": false,
-              "description": "Administrative username for RHPAM components",
-              "jsonPath": "$.spec.commonConfig.adminUser",
-              "type": "text"
+              "jsonPath": "$.spec.commonConfig",
+              "displayWhen": "default",
+              "visible": true,
+              "fields": [
+                {
+                  "label": "Username",
+                  "default": "adminUser",
+                  "required": false,
+                  "description": "Administrative username for RHPAM components",
+                  "jsonPath": "$.spec.commonConfig.adminUser",
+                  "type": "text"
+                },
+                {
+                  "label": "Password",
+                  "required": false,
+                  "description": "Administrative password to use instead of generating default value",
+                  "jsonPath": "$.spec.commonConfig.adminPassword",
+                  "type": "password"
+                }
+              ]
             },
             {
-              "label": "Password",
+              "label": "OpenShift admin credentials secret",
+              "type": "fieldGroup",
               "required": false,
-              "description": "Administrative password to use instead of generating default value",
-              "jsonPath": "$.spec.commonConfig.adminPassword",
-              "type": "password"
+              "jsonPath": "$.spec.commonConfig.secretAdminCredentials",
+              "displayWhen": "secret",
+              "visible": false,
+              "fields": [
+                {
+                  "label": "Secret configuration to use for Admin credentials",
+                  "required": false,
+                  "description": "If the secret is not present, kie-admin-credentials secret name will be used, if the secret is missing, one with the following username and password provided will be created",
+                  "jsonPath": "$.spec.commonConfig.secretAdminCredentials",
+                  "type": "text"
+                },
+                {
+                  "label": "Username",
+                  "default": "adminUser",
+                  "required": false,
+                  "description": "Administrative username to use in the secret if the secret is not already present, if leaved empty adminUser will be used",
+                  "jsonPath": "$.spec.commonConfig.adminUser",
+                  "type": "text"
+                },
+                {
+                  "label": "Password",
+                  "required": false,
+                  "description": "Administrative password to use in the secret if the secret is not already present, if leaved empty a generated one will be used",
```

```suggestion
                  "description": "Administrative password, if empty it will be generated, for trial environments the default password will be used.",
```

@desmax74 work with Clifton to outline the default username on the docs

---

ok